dowhy/causal_model.py

```diff
@@ -18,347 +19,378 @@
+		self._treatment = parse_state(treatment)
+		self._outcome = parse_state(outcome)
+		self._effect_modifiers = parse_state(effect_modifiers)
+		self._estimand_type = estimand_type
+		self._proceed_when_unidentifiable = proceed_when_unidentifiable
+		self._missing_nodes_as_confounders = missing_nodes_as_confounders
+		self.logger = logging.getLogger(__name__)
+
+		if graph is None:
+			self.logger.warning("Causal Graph not provided. DoWhy will construct a graph based on data inputs.")
+			self._common_causes = parse_state(common_causes)
+			self._instruments = parse_state(instruments)
+			if common_causes is not None and instruments is not None:
+				self._graph = CausalGraph(
+					self._treatment,
+					self._outcome,
+					common_cause_names=self._common_causes,
+					instrument_names=self._instruments,
+					effect_modifier_names=self._effect_modifiers,
+					observed_node_names=self._data.columns.tolist()
+				)
+			elif common_causes is not None:
+				self._graph = CausalGraph(
+					self._treatment,
+					self._outcome,
+					common_cause_names=self._common_causes,
+					effect_modifier_names = self._effect_modifiers,
+					observed_node_names=self._data.columns.tolist()
+				)
+			elif instruments is not None:
+				self._graph = CausalGraph(
+					self._treatment,
+					self._outcome,
+					instrument_names=self._instruments,
+					effect_modifier_names = self._effect_modifiers,
+					observed_node_names=self._data.columns.tolist()
+				)
+			else:
+				self._graph = None
+
+		else:
+			self.init_graph(graph=graph)
+			
+		self._other_variables = kwargs
+		self.summary()
+
+	def init_graph(self, graph):
+		'''
+		Initialize self._graph using graph provided by the user.
+
+		'''
+		# Create causal graph object
+		self._graph = CausalGraph(
+			self._treatment,
+			self._outcome,
+			graph,
+			effect_modifier_names=self._effect_modifiers,
+			observed_node_names=self._data.columns.tolist(),
+			missing_nodes_as_confounders = self._missing_nodes_as_confounders
+		)
+		self._common_causes = self._graph.get_common_causes(self._treatment, self._outcome)
+		self._instruments = self._graph.get_instruments(self._treatment,
+														self._outcome)
+		# Sometimes, effect modifiers from the graph may not match those provided by the user.
+		# (Because some effect modifiers may also be common causes)
+		# In such cases, the user-provided modifiers are used.
+		# If no effect modifiers are provided,  then the ones from the graph are used.
+		if self._effect_modifiers is None or not self._effect_modifiers:
+			self._effect_modifiers = self._graph.get_effect_modifiers(self._treatment, self._outcome)
+
+	def learn_graph(self, method_name="cdt.causality.graph.LiNGAM", render=False, view=False, *args, **kwargs):
+		'''
+		Learn causal graph from the data.
+		'''
+
+		if method_name is None:
+			#TODO add LiNGAM as default backdoor method, DirectLiNGAM as default LiNGAM method, add an informational message to show which method has been selected.
```

Comment:
is this todo done? if yes, then remove. otherwise, good idea to complete it. you can add a info message using logger.

Reply:
Can we just remove this line since the argument has a default value in the function definition itself?